src/execution/index/art/art.cpp

```diff
@@ -208,172 +207,7 @@ void ART::GenerateKeys(ArenaAllocator &allocator, DataChunk &input, vector<Key>
-				return;
-			}
-		}
-	}
-}
-
-void ART::ConstructAndMerge(IndexLock &lock, PayloadScanner &scanner, Allocator &allocator) {
-
-	auto payload_types = logical_types;
-	payload_types.emplace_back(LogicalType::ROW_TYPE);
-
-	ArenaAllocator arena_allocator(allocator);
-	vector<Key> keys(STANDARD_VECTOR_SIZE);
-
-	auto skipped_all_nulls = false;
-	auto temp_art = make_unique<ART>(this->column_ids, this->table_io_manager, this->unbound_expressions,
-	                                 this->constraint_type, this->db);
-
-	for (;;) {
-		DataChunk ordered_chunk;
-		ordered_chunk.Initialize(allocator, payload_types);
-		ordered_chunk.SetCardinality(0);
-		scanner.Scan(ordered_chunk);
-		if (ordered_chunk.size() == 0) {
-			break;
-		}
-
-		// get the key chunk and the row_identifiers vector
-		DataChunk row_id_chunk;
-		ordered_chunk.Split(row_id_chunk, ordered_chunk.ColumnCount() - 1);
-		auto &row_identifiers = row_id_chunk.data[0];
-
-		D_ASSERT(row_identifiers.GetType().InternalType() == ROW_TYPE);
-		D_ASSERT(logical_types[0] == ordered_chunk.data[0].GetType());
-
-		// generate the keys for the given input
-		arena_allocator.Reset();
-		GenerateKeys(arena_allocator, ordered_chunk, keys);
-
-		// we order NULLS FIRST, so we might have to skip nulls at the start of our sorted data
-		idx_t start_idx = 0;
-		FindFirstNotNullKey(keys, skipped_all_nulls, start_idx);
-
-		if (start_idx != 0 && IsPrimary()) {
-			throw ConstraintException("NULLs in new data violate the primary key constraint of the index");
-		}
-
-		if (!skipped_all_nulls) {
-			if (IsPrimary()) {
-				// chunk consists only of NULLs
-				throw ConstraintException("NULLs in new data violate the primary key constraint of the index");
-			}
-			continue;
-		}
-
-		// prepare the row_identifiers
-		row_identifiers.Flatten(ordered_chunk.size());
-		auto row_ids = FlatVector::GetData<row_t>(row_identifiers);
-
-		// construct the ART of this chunk
-		auto art = make_unique<ART>(this->column_ids, this->table_io_manager, this->unbound_expressions,
-		                            this->constraint_type, this->db);
-		auto key_section = KeySection(start_idx, ordered_chunk.size() - 1, 0, 0);
-		auto has_constraint = IsUnique();
-		Construct(keys, row_ids, art->tree, key_section, has_constraint);
-
-		// merge art into temp_art
-		if (!temp_art->MergeIndexes(lock, art.get())) {
-			throw ConstraintException("Data contains duplicates on indexed column(s)");
-		}
-	}
-
-	// NOTE: currently this code is only used for index creation, so we can assume that there are no
-	// duplicate violations between the existing index and the new data,
-	// so we do not need to revert any changes
-	if (!this->MergeIndexes(lock, temp_art.get())) {
-		throw ConstraintException("Data contains duplicates on indexed column(s)");
-	}
-}
-
-bool ART::Insert(IndexLock &lock, DataChunk &input, Vector &row_ids) {
+bool ART::Insert(IndexLock &lock, DataChunk &input, Vector &row_ids, bool skip_remove) {
```

Comment:
I'm not a big fan of adding the skip_remove here, since it also alters the `Index::Insert`
Would it make sense to have a `thread_local` attribute on the ART object and use it as a check there?  I would also consider adding a comment about why that check exists and return false.